modules/apps/27-interchain-accounts/types/metadata.go

```diff
@@ -9,3 +16,59 @@ func NewMetadata(version, controllerConnectionID, hostConnectionID, accAddress s
 		Address:                accAddress,
 	}
 }
+
+// ValidateControllerMetadata performs validation of the provided ICS27 controller metadata parameters
+func ValidateControllerMetadata(ctx sdk.Context, channelKeeper ChannelKeeper, connectionHops []string, metadata Metadata) error {
+	connection, err := channelKeeper.GetConnection(ctx, connectionHops[0])
+	if err != nil {
+		return err
+	}
+
+	if metadata.ControllerConnectionId != connectionHops[0] {
+		return sdkerrors.Wrapf(connectiontypes.ErrInvalidConnection, "expected %s, got %s", connectionHops[0], metadata.ControllerConnectionId)
+	}
+
+	if metadata.HostConnectionId != connection.GetCounterparty().GetConnectionID() {
+		return sdkerrors.Wrapf(connectiontypes.ErrInvalidConnection, "expected %s, got %s", connection.GetCounterparty().GetConnectionID(), metadata.HostConnectionId)
+	}
+
+	if metadata.Address != "" {
```

Comment:
If the address is an empty string should we be returning an error?

Reply:
No on OnChanOpenAck

Reply:
Right, I guess this is what your comment above was about as well